ha_test.go

```diff
@@ -7,150 +7,163 @@ import (
 	"testing"
 	"time"
 
-	"github.com/stretchr/testify/assert"
-	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/mock"
 	"github.com/stretchr/testify/suite"
 
 	"github.com/rockset/rockset-go-client"
 	"github.com/rockset/rockset-go-client/openapi"
 	"github.com/rockset/rockset-go-client/option"
 )
 
-func TestHA_Integration(t *testing.T) {
-	skipUnlessIntegrationTest(t)
+func (s *HASuite) TestHA_Integration() {
+	skipUnlessIntegrationTest(s.T())
 	const apikeyName = "ROCKSET_APIKEY_USE1A1" //nolint
 	apikey := os.Getenv(apikeyName)
 	if apikey == "" {
-		t.Skipf("skipping test as %s is not set", apikeyName)
+		s.T().Skipf("skipping test as %s is not set", apikeyName)
 	}
 
 	ctx := testCtx()
 
 	use1a1, err := rockset.NewClient(rockset.WithAPIServer("https://api.use1a1.rockset.com"),
 		rockset.WithAPIKey(apikey))
-	require.NoError(t, err)
+	s.NoError(err)
 
 	rs2, err := rockset.NewClient()
-	require.NoError(t, err)
+	s.NoError(err)
 
 	ha := rockset.NewHA(use1a1, rs2)
 	res, errs := ha.Query(ctx, "SELECT * FROM commons._events LIMIT 10")
-	assert.Len(t, errs, 0)
+	s.Len(errs, 0)
 
-	assert.Equal(t, "commons._events", (res.Collections)[0])
+	s.Equal("commons._events", (res.Collections)[0])
 }
 
-func (s *HaSuite) subFastest(test HaTest) {
-	ha := rockset.NewHA(test.first, test.second)
-	res, errs := ha.Query(s.ctx, "SELECT 1")
-	s.Nil(errs, errs)
-	s.Equal(test.fastID, *res.QueryId, "Response Query Id %d != %d", *res.QueryId, test.fastID)
-}
+func (s *HASuite) TestContextFail() {
+	a := createMock("1", time.Second, nil)
+	b := createMock("2", time.Second, nil)
+	ha := rockset.NewHA(a, b)
 
-func (s *HaSuite) subFail(test HaTest) {
-	ha := rockset.NewHA(test.first, test.second)
+	ctx := testCtx()
+	c, cancel := context.WithTimeout(ctx, time.Millisecond)
+	defer cancel()
 
-	_, errs := ha.Query(s.ctx, "SELECT 1")
-	s.Require().Len(errs, 1)
-	s.Equal(test.failError, errs[0].Error())
+	_, err := ha.Query(c, "SELECT 1")
+	s.Len(err, 1)
 }
 
-func (s *HaSuite) TestHATable() {
-	for _, test := range s.tests {
-		if test.fastID != "" {
-			s.Run(test.name, func() { s.subFastest(test) })
-		}
-
-		if test.failError != "" {
-			s.Run(test.name, func() { s.subFail(test) })
-		}
+func (s *HASuite) TestHA() {
+	for _, t := range s.tests {
+		mocks := createMocks(t.queries)
+		ha := rockset.NewHA(mocks...)
+		ctx := testCtx()
```

Comment:
you can move this outside the `for` loop as it returns the same context for all calls

Reply:
That's true currently, but seeing as it was called per test before, I'd rather leave it as is.
Anyone modifying that testCtx function can still assume it's behaving the same everywhere.
Either way I just don't like the function itself.